x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_color_property.js

```diff
@@ -117,11 +131,105 @@ export class DynamicColorProperty extends DynamicStyleProperty {
     return getColorRampStops(this._options.color);
   }
 
-  renderLegendHeader() {
+  renderRangeLegendHeader() {
     if (this._options.color) {
       return <ColorGradient colorRampName={this._options.color} />;
     } else {
       return null;
     }
   }
+
+  _renderStopIcon(color, isLinesOnly, isPointsOnly, symbolId) {
+    if (this.getStyleName() === VECTOR_STYLES.LABEL_COLOR) {
+      const style = { color: color };
+      return (
+        <EuiText size={'xs'} style={style}>
+          Tx
+        </EuiText>
+      );
+    }
+
+    if (isLinesOnly && this.getStyleName() === VECTOR_STYLES.LINE_COLOR) {
```

Comment:
The logic from this point downward contains a lot of duplication from `VectorIcon`. Maybe just render `VectorIcon` instead?